code/cycler/src/wattrex_battery_cycler/mid/mid_str/mid_str_node.py

```diff
@@ -0,0 +1,166 @@
+        self.db_iface.gen_meas: CyclerDataGenMeasC     = self.globlal_gen_meas.read()
+        self.db_iface.ext_meas: CyclerDataExtMeasC     = self.globlal_ext_meas.read()
+
+    def stop(self) -> None:
+        """Stop the node if it is not already closed .
+        """
+        self.db_iface.close_db_connection()
+        self.working_flag.clear()
+        self.status = SysShdNodeStatusE.STOP
+        log.critical(f"Stopping {current_thread().name} node")
+
+    def process_iteration(self) -> None:
+        """AI is creating summary for process_iteration
+        """
+        try:
+            # Syncronising shared data
+            self.sync_shd_data()
+            # Receive and write alarms
+            self.__receive_alarms()
+
+            if len(self.__new_raised_alarms)>0:
+                self.db_iface.write_new_alarm(alarms= self.__new_raised_alarms,
+                                              exp_id= self.__actual_exp_id)
+                self.__new_raised_alarms.clear()
+            # log.debug("+++++ After write alarams in db_iface object +++++")
+            ### Write measures and status changes
+            if self.db_iface.gen_meas.instr_id is not None:
+                self.db_iface.write_generic_measures(exp_id= self.__actual_exp_id)
+                log.critical("TEST COMMITING CHANGES")
+                self.db_iface.commit_changes()
+                self.db_iface.write_extended_measures(exp_id= self.__actual_exp_id)
+                self.db_iface.write_status_changes(exp_id= self.__actual_exp_id)
+                self.db_iface.meas_id += 1
+            if not self.str_reqs.is_empty():
+                # Ignore warning as receive_data return an object,
+                # which in this case must be of type DrvCanCmdDataC
+                command : MidStrCmdDataC = self.str_reqs.receive_data() # type: ignore
+                self.str_reqs.delete_until_last()
+                log.debug(f"Command to apply: {command.cmd_type.name}")
+                self.__apply_command(command)
+            # log.debug("+++++ Before commit changes in db_iface +++++")
+            # TIMEOUT added to detect if database connection was ended
+            func_timeout(TIMEOUT_CONNECTION, self.db_iface.commit_changes)
+            # log.debug("+++++ After commit changes in db_iface +++++")
+        except FunctionTimedOut as exc:
+            log.warning(("Timeout during commit changes to local database."
+                         f"Database connection will be restarted. {exc}"))
+            self.status = SysShdNodeStatusE.COMM_ERROR
+            self.db_iface.reset_db_connection()
+        except ConnectionError as exc:
+            self.status = SysShdNodeStatusE.COMM_ERROR
+            log.critical(exc)
+        except Exception as exc:
+            self.status = SysShdNodeStatusE.INTERNAL_ERROR
+            log.critical(f"Unexpected error in MID_STR_Node_c thread.\n{exc}")
+            # self.stop()
```

Comment:
Stop node if generic exception has been thrown.

Reply:
Working flag will be cleared